cpp/models/abm/person.cpp

```diff
@@ -256,5 +256,49 @@ bool Person::apply_mask_intervention(const Location& target)
     return true;
 }
 
+ScalarType Person::get_protection_factor(TimePoint t, VirusVariant virus, const GlobalInfectionParameters& params) const
```

Comment:
After some consideration, I believe that we should make this general such that the user can implement the effect of vaccines and prior infections if desired. For this, one should create a parameter and pass a function. Please open another issue for this.

Reply:
I have adjusted it to called a Parameter which then pass in a function defined by user. I think I will keep it as linear_piecewise_function for now.